data/src/main/java/io/homeassistant/companion/android/data/integration/IntegrationRepositoryImpl.kt

```diff
@@ -37,6 +38,46 @@ class IntegrationRepositoryImpl @Inject constructor(
         return localStorage.getString(PREF_WEBHOOK_ID) != null
     }
 
+    override suspend fun updateLocation(updateLocation: UpdateLocation) {
+        val updateLocationRequest = createUpdateLocation(updateLocation)
+        var wasSuccess = false
+        getUrls().forEach {
+            try {
+                wasSuccess =
+                    integrationService.updateLocation(it, updateLocationRequest).isSuccessful
+                return@forEach // If we get here update success!
+            } catch (e: Exception) {
+                // Ignore failure until we are out of URLS to try!
+            }
+        }
+
+        if (!wasSuccess) {
+            throw IntegrationException()
+        }
+    }
+
+    // https://developers.home-assistant.io/docs/en/app_integration_sending_data.html#short-note-on-instance-urls
+    private suspend fun getUrls(): Array<String> {
+        val retVal = ArrayList<String>()
+        val webhook = localStorage.getString(PREF_WEBHOOK_ID)
+
+        localStorage.getString(PREF_CLOUD_URL)?.let {
+            retVal.add(it)
+        }
+
+        localStorage.getString(PREF_REMOTE_UI_URL)?.let {
+            val base = if (it.last() == '/') it else "$it/"
```

Comment:
you can use okhttp to help you build this url 
okhttp3.HttpUrl
String.toHttpUrl()

Reply:
I migrated to using HttpUrl, but we still need the `/` check to ensure we don't get a double `/` and break the integration.

Reply:
A good unit test could help you assert this behaviour 🙄 

````
"https://www.home-assistant.io".toHttpUrl().newBuilder().addPathSegment("foo")
"https://www.home-assistant.io/".toHttpUrl().newBuilder().addPathSegment("foo")
"https://www.home-assistant.io/bar".toHttpUrl().newBuilder().addPathSegment("foo")
"https://www.home-assistant.io/bar/".toHttpUrl().newBuilder().addPathSegment("foo")
````

>https://www.home-assistant.io/foo
https://www.home-assistant.io/foo
https://www.home-assistant.io/bar/foo
https://www.home-assistant.io/bar/foo